tutorials/text/deduplication/semantic/semantic_e2e.ipynb

```diff
@@ -0,0 +1,843 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "9fd4cadb",
+   "metadata": {},
+   "source": [
+    "## Running as a single stage (e2e)\n",
+    "\n",
+    "The workflow does all of things behind the scenes\n",
+    "1. Read original dataset\n",
+    "2. Run Embedding Generation \n",
+    "3. Use KMeans to cluster the embeddings\n",
+    "4. Compute pairwise similarity inside each of the clusters\n",
+    "5. Identify duplicates based on `eps` provided (and `ranking_strategy`)\n",
+    "6. Remove duplites from the original dataset\n",
+    "\n",
+    "We also allow users to also run these steps indepedently, which can be seen in the tutorial semantic_step_by_step.ipynb [TODO: Can we add link to it?]\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "id": "11448dfb",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from nemo_curator.stages.deduplication.semantic import RankingStrategy\n",
+    "from nemo_curator.stages.text.deduplication.semantic import TextSemanticDeduplicationWorkflow\n",
+    "\n",
+    "workflow = TextSemanticDeduplicationWorkflow(\n",
+    "    input_path=input_path_to_data,\n",
+    "    output_path=output_path,\n",
+    "    cache_path=cache_path,\n",
+    "    perform_removal=True,\n",
+    "    # Embedding generation parameters\n",
+    "    text_field=\"text\",\n",
+    "    model_identifier=\"sentence-transformers/all-MiniLM-L6-v2\",\n",
+    "    embedding_max_seq_length=512,\n",
+    "    embedding_max_chars=None,\n",
+    "    embedding_pooling=\"mean_pooling\",\n",
+    "    embedding_model_inference_batch_size=256,\n",
+    "    # Semantic deduplication parameters\n",
+    "    n_clusters=100,  # this number can be much higher when the data is large\n",
+    "    # For large scale data we should use CURATOR_DEDUP_ID_STR if we are\n",
+    "    # also performing removal.\n",
+    "    id_field=\"id\",\n",
+    "    eps=0.01,\n",
+    "    # K-means clustering parameters\n",
+    "    ranking_strategy=RankingStrategy(metadata_cols=[\"cosine_dist_to_cent\"], ascending=True),\n",
+    "    pairwise_batch_size=1024,\n",
+    "    # ID generator parameters\n",
+    "    # For large scale data we should set use_id_generator to True if we are performing removal.\n",
+    "    use_id_generator=False,\n",
+    "    id_generator_state_file=None,\n",
+    "    # I/O parameters\n",
+    "    input_filetype=input_filetype,\n",
+    "    input_files_per_partition=1,\n",
+    "    output_filetype=output_filetype,\n",
+    "    verbose=True,\n",
+    "    clear_output=True,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "id": "7672bd73",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "2025-09-16 11:59:48,062\tINFO worker.py:1630 -- Using address 127.0.1.1:6379 set in the environment variable RAY_ADDRESS\n",
```

Comment:
Personal preference, we could clear these logs for the final commit, to help the tutorial look a bit cleaner. Ultimately up to you though, since the alternative argument is that they could be considered helpful for the user to know what to expect here.